homeassistant/components/generic_thermostat/climate.py

```diff
@@ -344,8 +392,11 @@ async def async_set_temperature(self, **kwargs):
         temperature = kwargs.get(ATTR_TEMPERATURE)
         if temperature is None:
             return
-        self._target_temp = temperature
+        self._presets[PRESET_NONE] = temperature
```

Comment:
I am not that deep into this but just to make sure I understand this correct:
If a target temperature is being set, the thermostat always is falling back to the `none` preset?
I can't really confirm that behavior so you might want to revert this change.

Reply:
If you are setting a custom temp, you are going out preset. Preset, as the name suggest, need to be read only.
For example if you using `PRESET_AWAY` (with 16°) but you want temporary set 18° if the thermostat keep the preset `PRESET_AWAY`, you cant set `PRESET_AWAY` directly to restore 16°